cradle-core/src/main/java/com/exactpro/cradle/utils/CompressionType.java

```diff
@@ -0,0 +1,112 @@
+    ZLIB {
+        public static final int BUFFER_SIZE = 4096;
+
+        @Override
+        public byte[] compress(byte[] data) throws CompressException {
+            try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream(data.length)) {
+                Deflater deflater = new Deflater();
+                deflater.setInput(data);
+                deflater.finish();
+                byte[] buffer = new byte[BUFFER_SIZE];
+                while (!deflater.finished()) {
+                    int count = deflater.deflate(buffer);
+                    outputStream.write(buffer, 0, count);
+                }
+                deflater.end();
+                return outputStream.toByteArray();
+            } catch (IOException e) {
+                throw new CompressException("Data can't be compressed", e);
+            }
+        }
+
+        @Override
+        public byte[] decompress(byte[] data) throws CompressException {
+            try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream(data.length)) {
+                Inflater inflater = new Inflater();
+                inflater.setInput(data);
+                byte[] buffer = new byte[BUFFER_SIZE];
+                while (!inflater.finished()) {
+                    int count = inflater.inflate(buffer);
+                    outputStream.write(buffer, 0, count);
+                }
+                inflater.end();
+                return outputStream.toByteArray();
+            } catch (IOException | DataFormatException e) {
+                throw new CompressException("Data can't be decompressed", e);
+            }
+        }
+    },
+    LZ4 {
+        private final LZ4Compressor COMPRESSOR = LZ4Factory.fastestInstance().fastCompressor();
+        private final LZ4FastDecompressor DECOMPRESSOR = LZ4Factory.fastestInstance().fastDecompressor();
+
+        @Override
+        public byte[] compress(byte[] data) {
+            var length = data.length;
+            var maxLength = COMPRESSOR.maxCompressedLength(length);
+            var compressed = new byte[Integer.BYTES + maxLength];
+            var newLength = COMPRESSOR.compress(data, 0, length, compressed, Integer.BYTES, maxLength);
+            var buffer = ByteBuffer.wrap(compressed);
+            buffer.putInt(0, length);
+            return Arrays.copyOf(compressed, Integer.BYTES + newLength);
+        }
+
+        @Override
+        public byte[] decompress(byte[] data) {
+            return DECOMPRESSOR.decompress(data, Integer.BYTES, ByteBuffer.wrap(data).getInt());
+        }
+    };
+
+    public abstract byte[] compress(byte[] data) throws CompressException;
+
+    public abstract byte[] decompress(byte[] data) throws CompressException;
+
+    // https://www.rfc-editor.org/rfc/rfc1950
+    // 78 01 - No Compression/low
+    // 78 9C - Default Compression - used in our case
+    // 78 DA - Best Compression
+    public static final byte ZLIB_CMF = 0x78;
+    public static final byte ZLIB_FLG = (byte) 0x9C;
```

Comment:
Are we sure that the LZ4 compression cannot produce the same set of bytes at the beginning?